### DISCUSSION TITLE
Clonable project structure with separate code locations

### DISCUSSION THREAD

**Discussion by Programermus** [dagster-io/dagster#19010] [Q&A]
Hi, i have a question about project structure.

It would be great for me if I could use a project structure like this:
```
.
└── Root (git repo)/
    ├── project_name/
    │   ├── code_location_1/
    │   │   ├── assets/
    │   │   │   ├── __init__.py
    │   │   │   ├── extract/
    │   │   │   │   ├── __init__.py
    │   │   │   │   ├── source_name_1.py
    │   │   │   │   └── source_name_2.py
    │   │   │   ├── transform
    │   │   │   └── load
    │   │   ├── resources
    │   │   └── sensors
    │   ├── code_location_2/
    │   │   ├── assets
    │   │   └── resources
    │   └── code_location_shared/
    │       ├── assets
    │       └── resources
    ├── project_name_tests/
    │   └── tests.py
    ├── .env
    ├── README
    └── .gitignore
```

Currently, I have implemented the project structure from: https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured , so what I would like to achieve (kind of) is to add an abstraction level (code_location_xx) within this structure, while still being able to use the cli commands like `dagster dev`. Would this be possible in a reasonable way?

Bonus question: within each code location it would be great if an asset defined in source_name_1.py would be structured in the asset group "extract" and asset key prefixed with "source_name_1". Can I achieve this?

**Comment by smackesey** [comment]
There's nothing to prevent you from organizing your project in this way, with the caveat that your `code_location_shared/assets` isn't going to work-- an asset can have only a single definition within a deployment, so if you pass the same `AssetsDefinition` into multiple code locations you will get an error.

>within each code location it would be great if an asset defined in source_name_1.py would be structured in the asset group "extract" and asset key prefixed with "source_name_1". Can I achieve this?

Check out [`load_assets_from_modules`](https://docs.dagster.io/_apidocs/assets#dagster.load_assets_from_modules)
